src/Resources/app/storefront/src/js/filter-category-select.plugin.js

```diff
@@ -105,6 +122,120 @@ export default class FilterCategorySelectPlugin extends FilterBasePlugin {
         this.listing.changeListing();
     }
 
+    /**
+     * @param {HTMLObjectElement} iconShell
+     * @param {string} removeClass
+     * @param {string} addClass
+     * @public
+     */
+    toggleCategoryListIconState(iconShell, removeClass, addClass) {
+
+        if (typeof (iconShell) !== undefined && iconShell !== null) {
+            iconShell.classList.remove(removeClass);
+            iconShell.classList.add(addClass);
+        }
+    }
+
+    /**
+     * @param {HTMLObjectElement} checkbox
+     * @public
+     */
+    onChangeCheckBox(checkbox) {
+        const subCategories = checkbox.parentNode.getElementsByClassName('sub-item');
+        const arrow = checkbox.parentNode.querySelector('#arrow');
+        const subCategory = arrow !== null ? arrow.nextElementSibling : null;
+
+        if (!checkbox.checked) {
+            Iterator.iterate(subCategories, (category) => {
+                category.querySelector('.filter-category-select-checkbox').checked = false;
+            });
+            return;
+        }
+
+        const categoryNames = checkbox.value.split('_');
+
+        if (categoryNames.length > 0) {
+            let categoryNameAsId = ''
+            Iterator.iterate(categoryNames, (name) => {
+                categoryNameAsId += categoryNameAsId === '' ? name : "_" + name;
+                document.getElementById(categoryNameAsId).checked = true;
+            });
+        }
+
+        this.toggleCategoryListIconState(arrow, 'down-arrow', 'up-arrow');
+        this.toggleSubcategoryDisplay(subCategory, 'hide-category-list-item', 'show-category-list-item');
+    }
+
+    /**
+     * @param {HTMLObjectElement} arrow
+     * @public
+     */
+    toggleCategoryListItemAndIconState(arrow) {
+        const subCategory = arrow !== null ? arrow.nextElementSibling : null;
+        const classList = arrow.getAttribute('class');
+        const isDropdownOpen = classList.indexOf('up-arrow') > -1;
+
+        if (isDropDownOpen) {
+            this.toggleCategoryListIconState(arrow, 'up-arrow', 'down-arrow');
+            this.toggleSubcategoryDisplay(subCategory, 'show-category-list-item', 'hide-category-list-item');
+            return;
+        }
+        this.toggleCategoryListIconState(arrow, 'down-arrow', 'up-arrow');
+        this.toggleSubcategoryDisplay(subCategory, 'hide-category-list-item', 'show-category-list-item');
+    }
+
+    /**
+     * @param {HTMLObjectElement} subCategory
+     * @public
+     */
+    siblingsCategories(subCategory) {
+        const siblingCategories = [];
+        let sibling = '';
+        if (subCategory !== null) {
+            siblingCategories.push(subCategory);
+            do {
+                sibling = siblingCategories[siblingCategories.length - 1].nextElementSibling;
+                siblingCategories.push(sibling !== null ? sibling : '');
+            }
+            while (sibling !== null);
```

Comment:
Please add the while block after the ending brace of the do block.

```suggestion
            do {
                sibling = siblingCategories[siblingCategories.length - 1].nextElementSibling;
                siblingCategories.push(sibling !== null ? sibling : '');
            } while (sibling !== null);
```